data/events/scripts/player.lua

```diff
@@ -86,6 +86,25 @@ function Player:onLookInShop(itemType, count)
 end
 
 function Player:onMoveItem(item, count, fromPosition, toPosition, fromCylinder, toCylinder)
+	-- Store Inbox
```

Comment:
I would probably opt for a source modification here instead of relying on 'onMoveItem'. An example would be to create a new class called 'StoreInbox' which mimics the behavior of the normal Inbox. It wouldn't let the player to put item(s) inside the store inbox, nor its child containers, and even replacing shields wouldn't work.